prov/sockets/src/sock_progress.c

```diff
@@ -662,12 +664,12 @@ static int sock_pe_process_rx_read(struct sock_pe *pe,
 
 		mr = sock_mr_verify_key(rx_ctx->domain,
 					pe_entry->pe.rx.rx_iov[i].iov.key,
-					(void *) (uintptr_t) pe_entry->pe.rx.rx_iov[i].iov.addr,
+					&pe_entry->pe.rx.rx_iov[i].iov.addr,
 					pe_entry->pe.rx.rx_iov[i].iov.len,
 					FI_REMOTE_READ);
 		if (!mr) {
 			SOCK_LOG_ERROR("Remote memory access error: %p, %lu, %" PRIu64 "\n",
-				       (void *) (uintptr_t) pe_entry->pe.rx.rx_iov[i].iov.addr,
+				       &pe_entry->pe.rx.rx_iov[i].iov.addr,
```

Comment:
The log entry is for printing the address that is stored in pe_entry->pe.rx.rx_iov[i].iov.addr (passed in by user). Doing '&' will give the address of an internal data structure.